pkg/registry/rbac/reconciliation/reconcile_role.go

```diff
@@ -203,7 +203,8 @@ func computeReconciledRole(existing, expected RuleOwner, removeExtraPermissions
 	switch {
 	case !removeExtraPermissions && len(result.MissingRules) > 0:
 		// add missing rules in the union case
-		result.Role.SetRules(append(result.Role.GetRules(), result.MissingRules...))
+		tmp := append([]rbacv1.PolicyRule{}, result.Role.GetRules()...)
+		result.Role.SetRules(append(tmp, result.MissingRules...))
```

Comment:
we deepcopy the role above (`result.Role = existing.DeepCopyRuleOwner()`), don't we own the content locally?